### ISSUE TITLE
possibly-used-before-assignment doesn't understand never-returning-functions

### ISSUE THREAD

**Issue by chrisburr** [pylint-dev/pylint#9627]
### Bug description

```python
# pylint: disable=missing-module-docstring,invalid-name
import sys

if input("Enter something or nothing: "):
    sys.exit()
else:
    a = 1
print(a)
```

### Command used

```shell
pylint test.py
```

### Pylint output

```python
************* Module test
test.py:8:6: E0606: Possibly using variable 'a' before assignment (possibly-used-before-assignment)
```

### Expected behavior

Branches which end in a function which never returns shouldn't be considered as possible causes of never defined variables.

### Pylint version

```shell
pylint 3.2.0
astroid 3.2.0
Python 3.11.7 | packaged by conda-forge | (main, Dec 23 2023, 14:43:09) [GCC 12.3.0]
```

**Comment by nickdrozd** [comment]
I don't think is this a false positive. Variable `a` really is possibly undefined, in the sense that there are branches in which it is used but not defined. For example, if `sys.exit` is tampered with, a `NameError` can be raised:

```python
import sys

sys.exit = lambda: None
if input("Enter something or nothing: "):
    sys.exit()
else:
    a = 1
print(a)
```

There is an easy way to silence this warning, namely to define `a` unconditionally:

```python
import sys

if input("Enter something or nothing: "):
    sys.exit()

a = 1
print(a)
```

**Comment by jacobtylerwalls** [comment]
I think the proposal is to use the existing option `never-returning-functions`, which I think is a good idea and simple fix. The risk of tampering is almost nil, and it's under the user's control, if they want to test more rigorously they can supply an empty array to that setting.

**Comment by jacobtylerwalls** [comment]
Looks like `never-returning-functions` is documented as only dealing with one option, and there's already a TODO statement in the code to reuse it in pylint 4 for other purposes, so that's maybe better deferred.

In the meantime, we can solve the reporter's issue with using the existing util `is_terminating_func`.

**Comment by chrisburr** [comment]
Thanks a lot for the quick fix!
